src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -20,20 +20,21 @@ public partial class Ping
         private const int MaxIpHeaderLengthInBytes = 60;
         private static bool SendIpHeader => OperatingSystem.IsMacOS();
         private static bool NeedsConnect => OperatingSystem.IsLinux();
+        private static bool IsOSXLike = OperatingSystem.IsMacOS() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS();
         [ThreadStatic]
         private static Random? t_idGenerator;
 
         private PingReply SendPingCore(IPAddress address, byte[] buffer, int timeout, PingOptions? options)
         {
-            PingReply reply = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) ?
+            PingReply reply = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) || IsOSXLike ?
                     SendIcmpEchoRequestOverRawSocket(address, buffer, timeout, options) :
                     SendWithPingUtility(address, buffer, timeout, options);
             return reply;
         }
 
         private async Task<PingReply> SendPingAsyncCore(IPAddress address, byte[] buffer, int timeout, PingOptions? options)
         {
-            Task<PingReply> t = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) ?
+            Task<PingReply> t = RawSocketPermissions.CanUseRawSockets(address.AddressFamily) || IsOSXLike ?
```

Comment:
Consider switching the order of these checks (`IsOSXLike || RawSocketPermissions.CanUseRawSockets(address.AddressFamily)`) so that if IsOSXLike is true this whole thing will evaporate into a const in tier 1.